packages/frontend/src/hooks/validation/useValidation.ts

```diff
@@ -34,14 +39,23 @@ export const useValidationMutation = (projectUuid: string) => {
     const { showError } = useErrorLogs();
     const { showToastSuccess } = useToaster();
 
-    return useMutation<ValidationResponse[], ApiError>({
+    return useMutation<ApiJobScheduledResponse['results'], ApiError>({
         mutationKey: ['validation', projectUuid],
         mutationFn: () => updateValidation(projectUuid),
-        onSuccess: () => {
-            queryClient.invalidateQueries({ queryKey: ['validation'] });
-            showToastSuccess({
-                title: 'Success! Validation complete.',
-            });
+        onSuccess: (data) => {
+            // Wait until validation is complete
+            pollJobStatus(data.jobId)
```

Comment:
and this is how you convert a sync method into a scheduler method with polling :kissing: :ok_hand:

Reply:
👏